Segment-Optimizely-X/Classes/SEGOptimizelyXIntegration.m

```diff
@@ -0,0 +1,84 @@
+//
+//  SEGOptimizelyXIntegration.m
+//  Pods
+//
+//  Created by ladan nasserian on 8/17/17.
+//
+//
+
+#import "SEGOptimizelyXIntegration.h"
+#import <Analytics/SEGIntegration.h>
+#import <Analytics/SEGAnalyticsUtils.h>
+#import <Analytics/SEGAnalytics.h>
+
+
+@implementation SEGOptimizelyXIntegration
+
+#pragma mark - Initialization
+
+- (instancetype)initWithSettings:(NSDictionary *)settings andOptimizelyClient:(OPTLYClient *)client withAnalytics:(SEGAnalytics *)analytics
+{
+    if (self = [super init]) {
+        self.settings = settings;
+        self.client = client;
+        self.analytics = analytics;
+    }
+
+    self.observer = [[NSNotificationCenter defaultCenter] addObserverForName:OptimizelyDidActivateExperimentNotification
+                                                                      object:nil
+                                                                       queue:nil
+                                                                  usingBlock:^(NSNotification *_Nonnull note) {
+                                                                      [self experimentDidGetViewed:note];
+                                                                  }];
+    return self;
+}
+
+- (void)identify:(SEGIdentifyPayload *)payload
+{
+    if (payload.userId) {
+        self.userId = payload.userId;
+    }
+}
+
+
+- (void)track:(SEGTrackPayload *)payload
+{
+    // Segment will default sending `track` calls with `anonymousId`s since Optimizely X does not alias known and unknown users
+    // https://developers.optimizely.com/x/solutions/sdks/reference/index.html?language=objectivec&platform=mobile#user-ids
+    NSString *segmentAnonymousId = [self.analytics getAnonymousId];
+    BOOL trackKnownUsers = [[self.settings objectForKey:@"trackKnownUsers"] boolValue];
+
+    if (trackKnownUsers && self.userId) {
```

Comment:
Prateek and I simplified our logic a bit: https://github.com/segment-integrations/analytics-android-integration-optimizelyx/pull/1/files#diff-0a51de86adf3eb1bbb6cfa913dac15a3R79